core/src/test/kotlin/api/RunsRouteIntegrationTest.kt

```diff
@@ -1151,6 +1154,203 @@ class RunsRouteIntegrationTest : AbstractIntegrationTest({
+                    jobConfigurations = JobConfigurations()
+                )
+
+                val obsoleteEvaluatorJob = dbExtension.fixtures.createEvaluatorJob(
+                    ortRunId = obsoleteOrtRun.id,
+                    configuration = EvaluatorJobConfiguration()
+                )
+
+                val obsoleteRuleViolations = listOf(
+                    OrtRuleViolation(
+                        "Rule-1-obsolete",
+                        Identifier(
+                            "Maven",
+                            "org.apache.logging.log4j.obsolete",
+                            "log4j-core",
+                            "2.14.0"
+                        ),
+                        "License-1-obsolete",
+                        "CONCLUDED",
+                        Severity.WARNING,
+                        "Message-1-obsolete",
+                        "How_to_fix-1-obsolete"
+                    ),
+                    OrtRuleViolation(
+                        "Rule-2-obsolete",
+                        Identifier(
+                            "Maven",
+                            "com.fasterxml.jackson.core.obsolete",
+                            "jackson-databind",
+                            "2.9.6"
+                        ),
+                        "License-2-obsolete",
+                        "DETECTED",
+                        Severity.ERROR,
+                        "Message-2-obsolete",
+                        "How_to_fix-2-obsolete"
+                    )
+                )
+
+                obsoleteRuleViolations.forEach {
+                    it.packageId?.let { identifier ->
+                        dbExtension.fixtures.createIdentifier(
+                            identifier
+                        )
+                    }
+                }
+
+                dbExtension.fixtures.evaluatorRunRepository.create(
+                    evaluatorJobId = obsoleteEvaluatorJob.id,
+                    startTime = Clock.System.now().toDatabasePrecision(),
+                    endTime = Clock.System.now().toDatabasePrecision(),
+                    violations = obsoleteRuleViolations
+                )
+
+                val response = superuserClient.get("/api/v1/runs/${ortRun.id}/rule-violations")
+
+                response shouldHaveStatus HttpStatusCode.OK
+
+                val ruleViolationsResponse = response.body<PagedResponse<RuleViolationWithIdentifier>>()
+
+                ruleViolationsResponse.data shouldHaveSize(4)
+
+                with(ruleViolationsResponse.pagination) {
+                    sortProperties shouldBe listOf(SortProperty("rule", SortDirection.ASCENDING))
+                    totalCount shouldBe 4
+                }
+
+                with(ruleViolationsResponse.data.first()) {
+                    val expectedFirstRuleViolation = ruleViolations[2]
+                    ruleViolation.howToFix shouldBe expectedFirstRuleViolation.howToFix
```

Comment:
Unfotunately, in this case, fixture that creates test DB entries operates on different object types than objects returned from API. 
Sure I can make changes to adjust it, but it's gonna be a bit of refactoring, and I don't know if it's worth it.

Reply:
And one of the mapping functions (`mapToModel` / `mapToApi`) cannot be used?

Reply:
You have to reorder the commits - the first commit already accesses functionality introduced in the second one.

However, what concerns me more is that there still seem to be some inconsistencies in the data model for rule violations:
* In ORT the `RuleViolation` class has a nullable identifier; so there can be violations not assigned to a package.
* In the ORT Server database, `RuleViolationTable` analogously has a nullable reference to the identifiers table.
* In ORT Server, the `RuleViolationService` IIUC does an inner join between rule violations and the identifiers table; so it will return only violations associated to a package. Its return type `RuleViolationWithIdentifier` reflects this.
* With your refactoring, you now make the identifier in `RuleViolationWithIdentier` nullable, which somehow contradicts the purpose of this class.

I think, the correct approach would be to change `RuleViolationService` to return all violations, no matter whether they have a package identifier or not. The return type should then be `OrtRuleViolation`, and `RuleViolationWithIdentifier` would no longer be needed.

Note that this change is likely to impact the UI.

Reply:
This will certainly impact the UI, as in PR #1195, `Identifier` is used extensively. For curation purposes, I _think_ we would need the identifier of the associated package, if it exists.

Additionally, please return also the `id` of the identifier, as we would need an id which is unique in some context (ORT run) in the UI, in order to create unique routes for the rule violations and show each in their own dedicated page. Currently, it is not possible, as no unique identifiers are provided by the endpoint.

Reply:
@oheger-bosch We're creating new entity anyway (API model)

Reply:
@oheger-bosch short answer: we don't. The first implementation doesn't use unique sub-routes (ie. pages) for vulnerabilities, just because we didn't have unique ids for them available. Instead, we show the details in the same table where we list the vulnerabilities, by using an expandable row model.

However, especially @sschuberth has expressed a need for showing more visual data for a vulnerability. There are some pretty decent libraries for this, but this would require us to put vulnerability details to their own sub-pages, and for that, we need unique ids for vulnerabilities.

This is a general pattern for the UI, so maybe I'll create a separate issue for it.

Reply:
@Etsija Can You please look into my UI changes in ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/rule-violations/index.tsx?

@oheger-bosch I've changed behavior of RuleViolations service to retrieve ones without package attached.